### PR TITLE
ESP32_2432S028R driver

### PR DESCRIPTION
**Pull request by nitroxgas**
Implementation of the ESP32_2432S028R driver; Added public-pool.io information and supplementary bottom screen.

### CODE REVIEW

File: src/monitor.cpp [hunk 1]
```diff
@@ -258,4 +269,46 @@ coin_data getCoinData(unsigned long mElapsed)
...
+        if (WiFi.status() != WL_CONNECTED) return pData;
+            
+        //Make first API call to get global hash and current difficulty
+        HTTPClient http;
+        http.setReuse(true);        
+        try {          
+          http.begin(String(getPublicPool)+btcString);  
+          int httpCode = http.GET();
+
+          if (httpCode > 0) {
```
Comment (romanmashta) [inline src/monitor.cpp:286]: http code should be HTTTP OK - 200
`if (httpCode == HTTP_CODE_OK)`

File: src/monitor.cpp [hunk 2]
```diff
@@ -52,6 +53,7 @@ void updateGlobalData(void){
             
         //Make first API call to get global hash and current difficulty
         HTTPClient http;
+        try {
```
Comment (romanmashta) [inline src/monitor.cpp:56]: Are you sure that this is correct way to handle errors here? I dont see that HTTPClient implementation raised any errors, nor DynamicJsonDocument. Deserialization errors usually handled with error code instead.

```
  DeserializationError error = deserializeJson(doc, resultString);
  if (error) {
```

**Comment by romanmashta** [comment]
Looks cool.

File: src/monitor.h [hunk 3]
```diff
@@ -26,6 +26,11 @@
 #define getFees "https://mempool.space/api/v1/fees/recommended"
 #define UPDATE_Global_min 2
 
+//API public-pool.io
+// https://public-pool.io:40557/api/client/btcString
+#define getPublicPool "https://public-pool.io:40557/api/client/" // +btcString
```
[the remark this answers is not in the document]
Comment (romanmashta) [reply src/monitor.h:31]: Maybe you can take pool URL from settings, and just stop querying getting pool data if it return 404 or non 200 HTTP error code on first updatePoolData execution? That will mean that user selected pool does not support that endpoint

File: src/monitor.cpp (hunk 2 above)
Comment (nitroxgas) [reply src/monitor.cpp:56]: During my tests, I found out that the ESP-WROOM-32 has no memory to process the https request and handle the display like the ESP32-S3, so the .GET function crashes at diffent points. As a result, a piece of memory get lost each time, until it freezes or restart the esp in a few minutes.
So this "try" have the intention to free the alocated memory if a error has occurred.
In time, I changed the TFT_eSprite alocation , creating and detroing it before any http request, so the Httpclient find enough memory to work.
But if there´s a better way to do this, I´ll be happy to learn.

File: src/monitor.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (nitroxgas) [reply src/monitor.h:31]: I will research other pools to see if they have a similar way of delivering this data, and then improve this compatibility part with other pools. This involves changing the code and images, but nothing too complicated.
I'm also trying to implement screen touch to switch between screens.

[the remark this answers is not in the document]
Comment (BitMaker-hub) [reply src/monitor.h:31]: Yes this is something I agree. But for now is ok. Let's go on

[the remark this answers is not in the document]
Comment (romanmashta) [reply src/monitor.h:31]: @BitMaker-hub Seems that part was merged to master. Can you merge it to dev too?

**Comment by BitMaker-hub** [comment]
oh... I see my fail here, should be on dev first

File: src/monitor.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (BitMaker-hub) [reply src/monitor.h:31]: done

**Pull request merged**